### PR TITLE
add prophet support

### PR DESCRIPTION
**Pull request by westfoxtrot**
##  Description

Add support for prophet PCB

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/westfoxtrot/prophet/config.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* If defined, GRAVE_ESC will always act as ESC when CTRL is held.
+ * This is userful for the Windows task manager shortcut (ctrl+shift+esc).
+ */
+#define GRAVE_ESC_CTRL_OVERRIDE
```
Comment (tzarc) [inline keyboards/westfoxtrot/prophet/config.h:38]: This should probably be removed, or moved into a personal keymap.

```suggestion
```

File: keyboards/westfoxtrot/prophet/rules.mk [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = STM32F072
+
+# Build Options
+#   comment out to disable the options.
+#
+
+
```
Comment (tzarc) [inline keyboards/westfoxtrot/prophet/rules.mk:8]:
```suggestion
```

File: keyboards/westfoxtrot/prophet/bootloader_defs.h [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/* Address for jumping to bootloader on STM32 chips. */
```
Comment (zvecr) [inline keyboards/westfoxtrot/prophet/bootloader_defs.h:1]: This file is not required and should be deleted.

File: keyboards/westfoxtrot/prophet/config.h [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+*/
+#define MATRIX_ROW_PINS { C13, B2, B1, A4, A3 }
+#define MATRIX_COL_PINS { A6, A7, B0, A9, A8, A14, A15, B3, B4, B5, B8, B7, B6, B9 }
+#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL, or CUSTOM_MATRIX */
+#define DIODE_DIRECTION COL2ROW
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCING_DELAY 5
```
Comment (zvecr) [inline keyboards/westfoxtrot/prophet/config.h:28]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/westfoxtrot/prophet/prophet.c [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+bool led_update_kb(led_t led_state) {
+  bool res = led_update_user(led_state);
+  if(res) {
+    // writePin sets the pin high for 1 and low for 0.
+    // In this example the pins are inverted, setting
+    // it low/0 turns it on, and high/1 turns the LED off.
+    // This behavior depends on whether the LED is between the pin
+    // and VCC or the pin and GND.
```
Comment (zvecr) [inline keyboards/westfoxtrot/prophet/prophet.c:15]:
```suggestion
```

File: keyboards/westfoxtrot/prophet/readme.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# prophet
+
+Keyboard Maintainer: [westfoxtrot](https://github.com/westfoxtrot)  
+Hardware Supported: All versions of the prophet PCB
+Hardware Availability: sold with prophet GB, [PCB Files](https://github.com/westfoxtrot/prophet_pcb)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make prophet:default
```
Comment (zvecr) [inline keyboards/westfoxtrot/prophet/readme.md:9]:
```suggestion
    make westfoxtrot/prophet:default
```

File: keyboards/westfoxtrot/prophet/readme.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+# prophet
+
+Keyboard Maintainer: [westfoxtrot](https://github.com/westfoxtrot)  
+Hardware Supported: All versions of the prophet PCB
+Hardware Availability: sold with prophet GB, [PCB Files](https://github.com/westfoxtrot/prophet_pcb)
```
Comment (zvecr) [inline keyboards/westfoxtrot/prophet/readme.md:5]:
```suggestion
* Keyboard Maintainer: [westfoxtrot](https://github.com/westfoxtrot)
* Hardware Supported: All versions of the prophet PCB
* Hardware Availability: sold with prophet GB, [PCB Files](https://github.com/westfoxtrot/prophet_pcb)
```

File: keyboards/westfoxtrot/prophet/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+#define _MA 0
+#define _F1 1
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [_MA] = LAYOUT_all(
+    KC_ESC,  KC_1,   KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,   KC_0,KC_MINS, KC_EQL,KC_GRV, KC_DEL,\
+    KC_TAB,   KC_Q,   KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,   KC_P,KC_LBRC,KC_RBRC,KC_BSPC,        \
+    KC_LCTRL,  KC_A,   KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,KC_SCLN,KC_QUOT,         KC_ENT,        \
+    KC_LSFT,  KC_BSLS,KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,KC_COMM, KC_DOT,KC_SLSH,KC_RSFT, MO(_F1),        \
```
Comment (fauxpark) [inline keyboards/westfoxtrot/prophet/keymaps/via/keymap.c:12]:
```suggestion
    KC_ESC,  KC_1,   KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,   KC_0,KC_MINS, KC_EQL,KC_GRV, KC_DEL,
    KC_TAB,   KC_Q,   KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,   KC_P,KC_LBRC,KC_RBRC,KC_BSPC,
    KC_LCTRL,  KC_A,   KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,KC_SCLN,KC_QUOT,         KC_ENT,
    KC_LSFT,  KC_BSLS,KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,KC_COMM, KC_DOT,KC_SLSH,KC_RSFT, MO(_F1),
```

File: keyboards/westfoxtrot/prophet/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    KC_LGUI,KC_LGUI,KC_LALT,                 KC_SPC,                        KC_RALT,KC_RGUI,KC_RCTL),
+
+  [_F1] = LAYOUT_all(
+    RESET,  KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9,  KC_F10, KC_F11,  KC_F12,KC_INS,KC_DEL, \
+    KC_CAPS,_______,_______,_______,_______,_______,_______,_______,KC_PSCR,KC_SLCK,KC_PAUS,KC_UP,_______,KC_DEL,         \
+    _______,KC_VOLD,KC_VOLU,KC_MUTE,KC_EJCT,_______,KC_PAST,KC_PSLS,KC_HOME,KC_PGUP,KC_LEFT,KC_RIGHT,        _______,         \
+    _______,_______,_______,_______,_______,_______,_______,KC_PPLS,KC_PMNS,KC_END,KC_PGDN,KC_DOWN,_______,_______,         \
```
Comment (fauxpark) [inline keyboards/westfoxtrot/prophet/keymaps/via/keymap.c:19]:
```suggestion
    RESET,  KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9,  KC_F10, KC_F11,  KC_F12,KC_INS,KC_DEL,
    KC_CAPS,_______,_______,_______,_______,_______,_______,_______,KC_PSCR,KC_SLCK,KC_PAUS,KC_UP,_______,KC_DEL,
    _______,KC_VOLD,KC_VOLU,KC_MUTE,KC_EJCT,_______,KC_PAST,KC_PSLS,KC_HOME,KC_PGUP,KC_LEFT,KC_RIGHT,        _______,
    _______,_______,_______,_______,_______,_______,_______,KC_PPLS,KC_PMNS,KC_END,KC_PGDN,KC_DOWN,_______,_______,
```

File: keyboards/westfoxtrot/prophet/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    _______,_______,_______,                _______,                        KC_STOP,_______,_______),
+
+  [2] = LAYOUT_all(
+    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,         \
+    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,        _______,         \
+    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,         \
```
Comment (fauxpark) [inline keyboards/westfoxtrot/prophet/keymaps/via/keymap.c:26]:
```suggestion
    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,        _______,
    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
```

File: keyboards/westfoxtrot/prophet/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    _______,_______,_______,                _______,                        _______,_______,_______),
+
+  [3] = LAYOUT_all(
+    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______, \
+    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,         \
+    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,        _______,         \
+    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,         \
```
Comment (fauxpark) [inline keyboards/westfoxtrot/prophet/keymaps/via/keymap.c:33]:
```suggestion
    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,        _______,
    _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,
```

**Comment by westfoxtrot** [comment]
@zvecr @tzarc

File: keyboards/westfoxtrot/prophet/prophet.h [hunk 12]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "quantum.h"
+
+#define ___ KC_NO
+
+#define LAYOUT( \
```
Comment (noroadsleft) [inline keyboards/westfoxtrot/prophet/prophet.h:7]: As the PCB supports multiple physical layouts, this should be named `LAYOUT_all`.

```suggestion
#define LAYOUT_all( \
```

File: keyboards/westfoxtrot/prophet/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,21 @@
+#include QMK_KEYBOARD_H
+
+#define _MA 0
+#define _F1 1
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [_MA] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/westfoxtrot/prophet/keymaps/default/keymap.c:8]:
```suggestion
  [_MA] = LAYOUT_all(
```

File: keyboards/westfoxtrot/prophet/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+    KC_ESC,  KC_1,   KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,   KC_0,KC_MINS, KC_EQL,KC_GRV, KC_DEL,
+    KC_TAB,   KC_Q,   KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,   KC_P,KC_LBRC,KC_RBRC,KC_BSPC,
+    KC_LCTRL,  KC_A,   KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,KC_SCLN,KC_QUOT,         KC_ENT,
+    KC_LSFT,  KC_BSLS,KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,KC_COMM, KC_DOT,KC_SLSH,KC_RSFT, MO(_F1),
+    KC_LGUI,KC_LGUI,KC_LALT,                 KC_SPC,                        KC_RALT,KC_RGUI,KC_RCTL),
+
+  [_F1] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/westfoxtrot/prophet/keymaps/default/keymap.c:15]:
```suggestion
  [_F1] = LAYOUT_all(
```

File: keyboards/westfoxtrot/prophet/keymaps/via/keymap.c [hunk 15]
```diff
@@ -0,0 +1,21 @@
+#include QMK_KEYBOARD_H
+
+#define _MA 0
+#define _F1 1
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [_MA] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/westfoxtrot/prophet/keymaps/via/keymap.c:8]:
```suggestion
  [_MA] = LAYOUT_all(
```

File: keyboards/westfoxtrot/prophet/keymaps/via/keymap.c [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+    KC_ESC,  KC_1,   KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,   KC_0,KC_MINS, KC_EQL,KC_GRV, KC_DEL,
+    KC_TAB,   KC_Q,   KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,   KC_P,KC_LBRC,KC_RBRC,KC_BSPC,
+    KC_LCTRL,  KC_A,   KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,KC_SCLN,KC_QUOT,         KC_ENT,
+    KC_LSFT,  KC_BSLS,KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,KC_COMM, KC_DOT,KC_SLSH,KC_RSFT, MO(_F1),
+    KC_LGUI,KC_LGUI,KC_LALT,                 KC_SPC,                        KC_RALT,KC_RGUI,KC_RCTL),
+
+  [_F1] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/westfoxtrot/prophet/keymaps/via/keymap.c:15]:
```suggestion
  [_F1] = LAYOUT_all(
```

File: keyboards/westfoxtrot/prophet/keymaps/via/keymap.c [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+    RESET,  KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9,  KC_F10, KC_F11,  KC_F12,KC_INS,KC_DEL,
+    KC_CAPS,_______,_______,_______,_______,_______,_______,_______,KC_PSCR,KC_SLCK,KC_PAUS,KC_UP,_______,KC_DEL,
+    _______,KC_VOLD,KC_VOLU,KC_MUTE,KC_EJCT,_______,KC_PAST,KC_PSLS,KC_HOME,KC_PGUP,KC_LEFT,KC_RIGHT,        _______,
+    _______,_______,_______,_______,_______,_______,_______,KC_PPLS,KC_PMNS,KC_END,KC_PGDN,KC_DOWN,_______,_______,
+    _______,_______,_______,                _______,                        KC_STOP,_______,_______),
+  };
```
Comment (noroadsleft) [inline keyboards/westfoxtrot/prophet/keymaps/via/keymap.c:21]: VIA currently requires at least four keymap layers by default.

```suggestion
  [2] = LAYOUT_all(
    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, 
    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, 
    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, 
    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, 
    _______, _______, _______,                            _______,                            _______, _______, _______  ),

  [3] = LAYOUT_all(
    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, 
    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, 
    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, 
    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, 
    _______, _______, _______,                            _______,                            _______, _______, _______  ),
  };
```

File: keyboards/westfoxtrot/prophet/config.h [hunk 18]
```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x21FF
```
Comment (drashna) [inline keyboards/westfoxtrot/prophet/config.h:6]: This vendor ID is already in use:
https://www.the-sz.com/products/usbid/index.php?v=0x21FF

File: keyboards/westfoxtrot/prophet/prophet.c [hunk 19]
```diff
@@ -0,0 +1,19 @@
...
+bool led_update_kb(led_t led_state) {
+  bool res = led_update_user(led_state);
+  if(res) {
+    writePin(B13, led_state.caps_lock);
+  }
+  return res;
+}
+
+__attribute__((weak)) layer_state_t layer_state_set_user(layer_state_t state) {
+  writePin(B12, (state & (1UL << 1)));
```
Comment (drashna) [inline keyboards/westfoxtrot/prophet/prophet.c:17]: You could also do this: 
```suggestion
  writePin(B12, layer_state_cmd(state, 1));
```

Comment (westfoxtrot) [reply keyboards/westfoxtrot/prophet/prophet.c:17]: `keyboards/westfoxtrot/prophet/prophet.c:17:17: error: implicit declaration of function 'layer_state_cmd'; did you mean 'layer_state_cmp'? [-Werror=implicit-function-declaration]`

I get this error with this change

Comment (fauxpark) [reply keyboards/westfoxtrot/prophet/prophet.c:17]: it should be `layer_state_cmp` as the error suggests.

**Comment by westfoxtrot** [comment]
@drashna @tzarc @zvecr @noroadsleft

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**